Doctrine/Phpcr/SitemapUrlInformationProvider.php

```diff
@@ -0,0 +1,160 @@
+ */
+
+namespace Symfony\Cmf\Bundle\SeoBundle\Doctrine\Phpcr;
+
+use Doctrine\ODM\PHPCR\DocumentManager;
+use Jackalope\Transport\Logging\LoggerInterface;
+use PHPCR\Query\QueryInterface;
+use Psr\Log\LoggerInterface as PsrLogger;
+use Symfony\Cmf\Bundle\CoreBundle\PublishWorkflow\PublishWorkflowChecker;
+use Symfony\Cmf\Bundle\SeoBundle\AlternateLocaleProviderInterface;
+use Symfony\Cmf\Bundle\SeoBundle\Extractor\ExtractorInterface;
+use Symfony\Cmf\Bundle\SeoBundle\Model\UrlInformation;
+use Symfony\Cmf\Bundle\SeoBundle\SeoPresentationInterface;
+use Symfony\Cmf\Bundle\SeoBundle\Sitemap\UrlInformationProviderInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\SecurityContextInterface;
+
+/**
+ * The PHPCR implementation of the sitemap route generator.
+ *
+ * @author Maximilian Berghoff <Maximilian.Berghoff@gmx.de>
+ */
+class SitemapUrlInformationProvider implements UrlInformationProviderInterface
+{
+    /**
+     * @var AlternateLocaleProviderInterface
+     */
+    protected $alternateLocaleProvider;
+
+    /**
+     * @var ExtractorInterface
+     */
+    protected $titleExtractor;
+
+    /**
+     * @var DocumentManager
+     */
+    private $manager;
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var SecurityContextInterface
+     */
+    private $publishWorkflowChecker;
+    /**
+     * @var string
+     */
+    private $defaultChanFrequency;
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+    /**
+     * @var SeoPresentationInterface
+     */
+    private $seoPresentation;
+
+    /**
+     * @param DocumentManager $manager
+     * @param RouterInterface $router
+     * @param SecurityContextInterface $publishWorkflowChecker
+     * @param string $defaultChanFrequency
+     * @param PsrLogger $logger
+     * @param SeoPresentationInterface $seoPresentation
+     */
+    public function __construct(
+        DocumentManager $manager,
+        RouterInterface $router,
+        SecurityContextInterface $publishWorkflowChecker,
+        $defaultChanFrequency,
+        PsrLogger $logger,
+        SeoPresentationInterface $seoPresentation
+    ) {
+        $this->manager = $manager ;
+        $this->router = $router;
+        $this->publishWorkflowChecker = $publishWorkflowChecker;
+        $this->defaultChanFrequency = $defaultChanFrequency;
+        $this->logger = $logger;
+        $this->seoPresentation = $seoPresentation;
+    }
+
+    /**
+     * {@inheritDocs}
+     */
+    public function generateRoutes()
+    {
+        $routeInformation = array();
+
+        $contentDocuments = $this->manager->createQuery(
+            "SELECT * FROM [nt:unstructured] WHERE (visible_for_sitemap = true)",
+            QueryInterface::JCR_SQL2
+        )->execute();
+
+        foreach ($contentDocuments as $document) {
+            if (!$this->publishWorkflowChecker->isGranted(array(PublishWorkflowChecker::VIEW_ATTRIBUTE), $document)) {
```

Comment:
we really should do https://github.com/symfony-cmf/CoreBundle/issues/126 at some point - then you could ask the publish workflow checker to add this to the query.